src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1726,7 +1752,10 @@ class AliasExpressionSegment(BaseSegment):
             ),
             # just a table alias
             Ref("SingleIdentifierGrammar"),
-            exclude=Ref("JoinTypeKeywords"),
+            exclude=[
+                Ref.keyword("LATERAL"),
+                Ref("JoinTypeKeywords"),
+            ],
```

Comment:
On the past we've just used this:

```suggestion
            exclude=OneOf(
                Ref.keyword("LATERAL"),
                Ref("JoinTypeKeywords"),
            ),
```

Does that negate the need for the iterable changes above? Or still think those are better?